src/vizier/services/query_broker/script_runner/script_source.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package scriptrunner
+
+import (
+	"context"
+
+	"github.com/nats-io/nats.go"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+
+	"px.dev/pixie/src/shared/cvmsgspb"
+	"px.dev/pixie/src/utils/shared/k8s"
+	"px.dev/pixie/src/vizier/services/metadata/metadatapb"
+)
+
+// Names of [Source]s
+const (
+	CloudSourceName     = "cloud"
+	ConfigMapSourceName = "configmaps"
+)
+
+// DefaultSources is a list of sources enabled by default
+var DefaultSources = []string{CloudSourceName}
+
+// A Source provides an initial set of cron scripts and sends incremental updates to that set
+type Source func(ctx context.Context, updateCb func(*cvmsgspb.CronScriptUpdate)) (initial map[string]*cvmsgspb.CronScript, unsubscribe func(), err error)
```

Comment:
Sorry for the delay! I had some more high level thoughts while I was reviewing this:
The `Source` type got a little confusing (lots of nested funcs, funcs that take funcs as args, etc).
I'm thinking this sort of architecture might look cleaner/readable:

```
type interface Source {
    GetInitialScripts() map[string]*cvmsgspb.CronScript
    Stop() error
    Start() error // does setupSource
}
```

which `CloudSource` and `ConfigMapSource` then implement. Then script runner can just iterate through the sources and call `source.GetInitialScripts()`, `source.Stop()`, etc when needed.

And rather than having an `updateCb`, which the sources call with any cronscript updates, what do you think about instead passing an `updateCh` channel to `Sources` here?
And then CloudSource and ConfigMapSource would add any new updates to `updateCh`, which `scriptrunner` can listen on.

Let me know what you think!

Reply:
This makes sense to me, we'll work on the refactor.